pkg/trace/agent/agent.go

```diff
@@ -391,20 +391,34 @@ func (a *Agent) setPayloadAttributes(p *api.Payload, root *pb.Span, chunk *pb.Tr
 		p.TracerPayload.Env = traceutil.GetEnv(root, chunk)
 	}
 	if p.TracerPayload.AppVersion == "" {
-		p.TracerPayload.AppVersion = traceutil.GetAppVersion(root, chunk)
+		p.TracerPayload.AppVersion = version.GetAppVersionFromTrace(root, chunk)
 	}
 }
 
-// processedTrace creates a ProcessedTrace based on the provided chunk and root.
-func processedTrace(p *api.Payload, chunk *pb.TraceChunk, root *pb.Span) *traceutil.ProcessedTrace {
-	return &traceutil.ProcessedTrace{
+// processedTrace creates a ProcessedTrace based on the provided chunk, root, containerID, and agent config.
+func processedTrace(p *api.Payload, chunk *pb.TraceChunk, root *pb.Span, containerID string, conf *config.AgentConfig) *traceutil.ProcessedTrace {
+	pt := &traceutil.ProcessedTrace{
 		TraceChunk:             chunk,
 		Root:                   root,
 		AppVersion:             p.TracerPayload.AppVersion,
 		TracerEnv:              p.TracerPayload.Env,
 		TracerHostname:         p.TracerPayload.Hostname,
 		ClientDroppedP0sWeight: float64(p.ClientDroppedP0s) / float64(len(p.Chunks())),
+		GitCommitSha:           version.GetGitCommitShaFromTrace(root, chunk),
+	}
+	// TODO: We should find a way to not repeat container ID resolution downstream in the stats writer.
+	// We will first need to deprecate the `enable_cid_stats` feature flag.
```

Comment:
nit: This comment feels like it's in the wrong place. The container ID is passed into this function which means you are doing the resolution elsewhere. And that place should probably have this comment. Also is this something you should address in this PR?

Reply:
Gotcha - I'll revise or remove. There is a corresponding comment in the stats writer where the resolution is taking place.

I spoke about whether we should address this with @knusbaum and we agreed that it would be OK to hold off for now. We need to figure out if it's safe to enable container ID aggregation in stats. If so, then we can do a larger refactor.

Reply:
I think this comment can be improved simply by saying "container tags resolution" rather than "container ID resolution".